lib/BUCK

```diff
@@ -0,0 +1,39 @@
+
+rust_library(
+    name = "jj-lib",
```

Comment:
Mini-Nit: Use the directory name for the target, as it is a Google adjacent project. As far as I observed only Meta use the `cell//project/lib:lib` label name as their chosen style.

Reply:
We can't do this because the `name` for `rust_library` is what becomes the crate name, so it has to be the same for imports to keep working.

However, adding the `:lib` alias fixes this, and it also means that running `buck2 build //jj-lib:jj-lib` can be shortented to `build //jj-lib` &mdash; when the target has the same name as the last component of a package path, it becomes the default target for that package.

Reply:
> However, adding the `:lib` alias above fixes this, but more importantly it means that running `buck2 build //lib:jj-lib` can be shortented to `buck2 build //lib`

FTR, I wanted it and a alias solves it fine.

Reply:
For posterity, you can also use `crate` to override the crate name: https://buck.build/rule/rust_library.html#crate